kimchi/src/circuits/expr.rs

```diff
@@ -1082,23 +1071,14 @@ impl<'a, F: FftField> EvalResult<'a, F> {
                     evals: es2,
                 },
             ) => {
-                let scale1 = (d1 as usize) / (res_domain.0 as usize);
-                assert!(scale1 != 0);
                 let scale2 = (d2 as usize) / (res_domain.0 as usize);
                 assert!(scale2 != 0);
 
-                let n = res_domain.1.size();
-                let v: Vec<_> = (0..n)
-                    .into_par_iter()
-                    .map(|i| {
-                        es1.evals[(scale1 * i + (d1 as usize) * s1) % es1.evals.len()]
-                            + es2.evals[(scale2 * i + (d2 as usize) * s2) % es2.evals.len()]
-                    })
-                    .collect();
-
                 Evals {
                     domain: res_domain.0,
-                    evals: Evaluations::<F, D<F>>::from_vec_and_domain(v, res_domain.1),
+                    evals: es1.to_subdomain_unsafe(res_domain.1, (d1 as usize) * s1, |eval, i| {
+                        eval + es2.evals[(scale2 * i + (d2 as usize) * s2) % es2.evals.len()]
```

Comment:
It's a pity that we still have to manually do this here.

Reply:
Maybe there's a way to make a "merging" helper?

Reply:
yeah it'd be nicer to have a function that clearly states what it's doing instead